### PR TITLE
Remove dependency on Python 3.6 for publish-scripts

### PR DESCRIPTION
**Pull request by michaelpeng36**
### Issue describing the changes in this PR

Removes the dependency on Python 3.6 for generating .DEB packages. Specifically, `platform.linux_distribution()` is deprecated and is replaced by `distro.name()`, as detailed by [this link](https://stackoverflow.com/questions/49554443/platform-linux-distribution-deprecated-what-are-the-alternatives).

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [x] Otherwise: Backport tracked by issue/PR #2874 
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by ejizba — COMMENTED** [review]
cc @davidmrdavid because I only have surface level python knowledge

File: publish-scripts/driver.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#! /usr/bin/env python3.6
+#! /usr/bin/env python3.8
```
[the remark this answers is not in the document]
Comment (michaelpeng36) [reply publish-scripts/driver.py:1]: Thanks for the catch, Eric. I was thinking this should actually be `/usr/bin/env python3`, which defaults to the latest Python 3 version. What do you think?

File: publish-scripts/driver.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
...
+import distro
```
[the remark this answers is not in the document]
Comment (michaelpeng36) [reply publish-scripts/driver.py:2]: Yes, that is correct. I have removed the manual installation of `wget` and replaced it with `pip install -r requirements.txt`.

**Review by ejizba — APPROVED** [review]

File: publish-scripts/driver.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ejizba) [reply publish-scripts/driver.py]: Probably fine 🤷‍♂️

**Pull request merged**